Source/SpacePark-API/DataAccess/APICollector.cs

```diff
@@ -0,0 +1,309 @@
+                var httpresponse = httpclient.GetAsync(url).Result;
+                httpresponse.EnsureSuccessStatusCode();
+                jsonResult = httpresponse.Content.ReadAsStringAsync().Result;
+            }
+
+            if (!string.IsNullOrWhiteSpace(jsonResult))
+            {
+                var parsedShip = JsonConvert.DeserializeObject<SpaceShip>(jsonResult);
+                var resultobject = JsonConvert.DeserializeObject<JObject>(jsonResult);
+                foreach (var property in resultobject.Properties())
+                    if (property.Name == "length")
+                        parsedShip.ShipLength = property.Value.ToString().Replace(",", "").Replace(".", "");
+                return parsedShip;
+            }
+
+            throw new Exception("Parse was empty; Is the URL in correct format?");
+        }
+
+        #endregion
+
+        public static Person ParseUser(string name)
+        {
+            name = System.Web.HttpUtility.UrlDecode(name); //Decode URL if it contains spacebar. This is to ensure it is always correct
+            //-no matter what encoding is used. Most common is percent20-encoding, but could also be plus-encoding(so that spacebars-
+            //will be replaced with + signs)
+            var foundUser = false;
+            var user = new Person();
+            for (var i = 1; i <= 9; i++)
+                if (!foundUser)
+                {
+                    var tempUsers =
+                        ReturnUsersFromList("https://swapi.dev/api/people/?page=" + i);
+                    foreach (var s in tempUsers)
+                    {
+                        if (s.Name.ToLower().Replace(" ", "").Replace("-", "") != name.ToLower().Replace(" ","").Replace("-","")) continue;
+                        foundUser = true;
+                        user = s;
+                        break;
+                    }
+                }
+                else
+                {
+                    break;
+                }
+
+            return foundUser ? user : null;
+        }
+
+        public static Person ParseUserAsync(string name) //This contains a threadstart for the private corresponding method
+        {
+            var user = new Person();
+            var thread = new Thread(() => { user = ParseUser(name); });
+            thread.Start();
+            thread.Join(); //By doing join it will wait for the method to finish
+            return user;
+        }
+
+        public static SpaceShip ParseShip(string model)
+        {
+            model = System.Web.HttpUtility.HtmlDecode(model);
+            
+            var foundShip = false; //A check to verify a ship has been found by name
+            var
+                ship = new SpaceShip(); //Had to initialize because code didn't realize it was bound to the foundShip bool and i got build errors.
+            for (var i = 1; i <= 4; i++)
```

Comment:
Number 4! [Magic Number](https://en.wikipedia.org/wiki/Magic_number_(programming))
First I had no idea what this number was for!
I'd give it an actual variable name so for example *pageAmount*

The other issue I see with this is why is it hard coded? what happens if SWApi adds a 5th page? you will have to remember about this project come in and **find** it in the project and manually change it.. Why not take advantage of the SWApi and use the *Next* attribute from the api. Make a method that gets the next until its null. This way you it will always work no matter if they have removed or added new ships :)

Hope this helps!